py/picca/cf.py

```diff
@@ -158,27 +158,28 @@ def compute_xi(healpixs):
                     (delta1.fiberid <= 500 and delta2.fiberid <= 500) or
                     (delta1.fiberid > 500 and delta2.fiberid > 500)))
                 if ang_correlation:
-                    (rebin_weight, rebin_xi, rebin_r_par, rebin_r_trans,
-                     rebin_z, rebin_num_pairs) = compute_xi_forest_pairs(
+                    compute_xi_forest_pairs_fast(
                          delta1.z, 10.**delta1.log_lambda,
                          10.**delta1.log_lambda, delta1.weights, delta1.delta,
                          delta2.z, 10.**delta2.log_lambda,
                          10.**delta2.log_lambda, delta2.weights, delta2.delta,
-                         ang, same_half_plate)
+                         ang, same_half_plate, 
+                         weights, xi, r_par, r_trans, z, num_pairs)
                 else:
-                    (rebin_weight, rebin_xi, rebin_r_par, rebin_r_trans,
-                     rebin_z, rebin_num_pairs) = compute_xi_forest_pairs(
+                    compute_xi_forest_pairs_fast(
                          delta1.z, delta1.r_comov, delta1.dist_m,
-                         delta1.weights, delta1.delta, delta2.z, delta2.r_comov,
-                         delta2.dist_m, delta2.weights, delta2.delta, ang,
-                         same_half_plate)
-
-                xi[:len(rebin_xi)] += rebin_xi
-                weights[:len(rebin_weight)] += rebin_weight
-                r_par[:len(rebin_r_par)] += rebin_r_par
-                r_trans[:len(rebin_r_trans)] += rebin_r_trans
-                z[:len(rebin_z)] += rebin_z
-                num_pairs[:len(rebin_num_pairs)] += rebin_num_pairs.astype(int)
+                         delta1.weights, delta1.delta, 
+                         delta2.z, delta2.r_comov, delta2.dist_m, 
+                         delta2.weights, delta2.delta, 
+                         ang, same_half_plate, 
+                         weights, xi, r_par, r_trans, z, num_pairs)
+
+                #xi[:len(rebin_xi)] += rebin_xi
```

Comment:
why are these commented? I take it the values are now updated in the functions themselves, aren't they? If that is correct, maybe we should remove these lines

Reply:
I left them because I also left the old versions of the functions and I would be easy to come back.

Reply:
ok, so maybe we want to add a comment specifying that these are deprecated commands (also for the older functions) that will be removed later